contracts/lib/DepositContract.sol

```diff
@@ -1,13 +1,13 @@
 // SPDX-License-Identifier: AGPL-3.0
-pragma solidity 0.8.15;
+pragma solidity 0.8.17;
 
-import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 
 /**
  * This contract will be used as a helper for all the sparse merkle tree related functions
  * Based on the implementation of the deposit eth2.0 contract https://github.com/ethereum/consensus-specs/blob/dev/solidity_deposit_contract/deposit_contract.sol
  */
-contract DepositContract is Initializable {
+contract DepositContract is ReentrancyGuardUpgradeable {
```

Comment:
q: why is `ReentrancyGuardUpgradeable` here, while  `__ReentrancyGuard_init()` and usage of `nonReentrant` are in `PolygonZkEVMBridge.sol` ?

Reply:
It was here just here just to be clear the hierarchy, i want to be the storage slots in that order:
Initializable --> ReentrancyGuardUpgradeable --> DepositContract --> EmergencyManager --> PolygonZkEVMBridge
But i could swap it, it's true, i could do something like:
DepositContract --> Initializable -->  ReentrancyGuardUpgradeable --> EmergencyManager --> PolygonZkEVMBridge

Reply:
Ok, The only thing I was thinking about: what if the `DepositContract` is reused in the future from another contract and someone forget to put in `__ReentrancyGuard_init()`. Maybe also put a comment about this in the code of `DepositContract`.

Reply:
> it's a goodidea, but, i think with the OZ lib that checks the storage slots when upgrade will be enough

what OZ library checks for storage slots on each upgrade? If it's the proxy contract, I think it only checks a rollback check when the proxy is UUPS. But in your case, transparent upgradeable proxies are used.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Upgrade.sol#L75-L89

Reply:
Unless this is what you mean https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/6a1969f54021cbc04287fb81a22449a9348893a7/packages/core/src/storage/index.ts#L22-L37

If I understand this correctly, `upgrades.upgradeProxy()` allows you to assert some new storage layout that is provided by the upgrader as an option?

Reply:
I din't know exactly, where in the OZ code, but here [upgradeProxy](https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades#upgrade-proxy) This function does an storage check ( as you can see there's this option unsafeSkipStorageCheck)